esp_encrypted_img/include/esp_encrypted_img.h

```diff
@@ -68,6 +68,10 @@ typedef void *esp_decrypt_handle_t;
 
 typedef struct {
 #if defined(CONFIG_PRE_ENCRYPTED_OTA_USE_RSA)
+#if defined(CONFIG_PRE_ENCRYPTED_RSA_USE_DS)
+    void *ds_data;                         /*!< Data context for DS */
```

Comment:
This could be purely dynamic as well - if `ds_data` is set then use DS peripheral approach else tradition RSA approach. Then we can remove `CONFIG_PRE_ENCRYPTED_RSA_USE_DS`

Reply:
I am not sure about making it dynamic and creating this fallback. If the choice is made during runtime dynamically, then it might be possible (intentionally or not) to fall back to a less secure option. If the choice is compile time, then we can guarantee the security, and if it fails the OTA will not succeed. Let me know what are your thoughts.